src/App.tsx

```diff
@@ -1,14 +1,66 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
-import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
+import { SortType, TodoFilter } from './components/TodoFilter';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { TodoModal } from './components/TodoModal';
+
+const todosFromServer = getTodos().then((todos: Todo[]) => todos);
 
 export const App: React.FC = () => {
+  const [loading, setLoading] = useState(false);
+  const [openTodo, setOpenTodo] = useState<Todo | null>(null);
+  const [currentTodos, setCurrentTodos] = useState<Todo[]>([]);
+  const [currentSelectedFilter, setCurrentSelectedFilter] = useState(
+    SortType.All,
+  );
+  const [currentQuery, setCurrentQuery] = useState('');
+
+  const onSelectedFilter = (filter: SortType) => {
+    setCurrentSelectedFilter(filter);
+  };
+
+  const onQuery = (query: string) => {
+    setCurrentQuery(query);
+  };
+
+  const TodoOpen = {
+    openTodo,
+    setTodo: setOpenTodo,
+  };
+
+  useEffect(() => {
+    setLoading(true);
+    todosFromServer.then(todos => {
```

Comment:
```suggestion
    getTodos().then(setTodos)
```
and  filter the array like that 
`const filteredTodos = getFilteredTodos(todos, query, currentSelectedFilter)`